src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -476,7 +476,10 @@ private static ThreeState TryGetBestResult(ArrayBuilder<MemberResolutionResult<T
                     && !isMethodGroupConversion
                     // Function pointer type symbols don't have named parameters, so we just want to report a general mismatched parameter
                     // count instead of name errors.
-                    && !(firstSupported.Member is FunctionPointerMethodSymbol _))
+                    // The only case we want this block to be entered for function pointers is if it's called with a named argument.
+                    // We have a special error for it (ERR_FunctionPointersCannotBeCalledWithNamedArguments) reported
+                    // in ReportNoCorrespondingNamedParameter.
+                    && (firstSupported.Member is not FunctionPointerMethodSymbol || firstSupported.Result.Kind == MemberResolutionKind.NoCorrespondingNamedParameter))
```

Comment:
>(firstSupported.Member is not FunctionPointerMethodSymbol || firstSupported.Result.Kind == MemberResolutionKind.NoCorrespondingNamedParameter) [](start = 23, length = 142)

I think the code would be easier to follow and maintain if we simply moved the check for function pointer and the resolution kind into a separate ```if``` branch and explicitly reported ```diagnostics.Add(ErrorCode.ERR_FunctionPointersCannotBeCalledWithNamedArguments, location)``` from there without making any changes to ReportNoCorrespondingNamedParameter.